[louistaii] iP

louistaii opened the pull request.

louistaii committed:
Add greeting

louistaii committed:
Add task list

Added function to insert tasks, mark and unmark task as done, and view all tasks.

louistaii committed:
Add multiple task types

now supports different task types (todo, deadline, event) and their respective dates

louistaii committed:
Bug fix and improvements

fixed looping task count. improved comparison of strings

JudHoka commented on src/main/java/Task.java:
```diff
@@ -16,13 +16,13 @@ public String getStatusIcon() {
         return (isDone ? "X" : " "); // mark done task with X
     }
     public String getTypeIcon() {
-        if (type == "todo"){
```
Almost pointed out that "==", great initiative with "equals"!

JudHoka commented on src/main/java/casio.java:
```diff
@@ -16,18 +17,50 @@ public static void exit(){
 
-    public static void print_tasks(Task[] task_array){
+    public static void print_tasks(Task[] task_array, int task_number){
         for (int i = 0; i < task_array.length; i++) {
```
Maybe you wanna try reducing the nested statements so that it is more understandable (avoid arrowhead thing), try grouping them in methods.

JudHoka commented on src/main/java/casio.java:
```diff
@@ -1,12 +1,13 @@
 public class casio{
+
     public static void main(String[] args) {
         String greeting = "____________________________________________________________"
```
Maybe make it a global variable instead of putting it into a main

JudHoka commented on src/main/java/casio.java:
```diff
@@ -16,18 +17,50 @@ public static void exit(){
+
+    public static void get_input(){
         Scanner scanner = new Scanner(System.in);
         Task[] task_array =  new Task[100]; //array to keep track of tasks
```
Avoid magic numbers (the 100)

theertha120 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,30 @@
+{
+    protected String description;
+    protected boolean isDone;
+    protected String type;
```
You can replace 'type' with a better variable name that is more informative such as 'commandType'

JudHoka commented on src/main/java/casio.java:
```diff
@@ -40,30 +73,51 @@ public static void get_task(){
-                print_tasks(task_array);
+                print_tasks(task_array, task_number);
             }
             else if (input.startsWith("mark ")){
```
Some idea I had is to make a method for mark and unmark, and just call the method everytime needed

JudHoka reviewed (commented):
Overall, awesome job bro!

theertha120 commented on src/main/java/casio.java:
```diff
@@ -0,0 +1,127 @@
+    public static void print_tasks(Task[] task_array, int task_number){
+        for (int i = 0; i < task_array.length; i++) {
+            if (task_array[i]!=null) {
+                String status_icon = task_array[i].getStatusIcon();
```
The variable names should be in camel case, so you can replace 'status_icon' with 'statusIcon', 'task_number' with 'taskNumber', and you can do so wherever applicable

theertha120 commented on src/main/java/casio.java:
```diff
@@ -0,0 +1,127 @@
+        System.out.println(bye);
+    }
+
+    public static void add_task(Task[] task_array, String name, String type, int task_number, String details){
```
Method names should be in camel case, so you can replace 'add_task' with 'addTask' and you can do so for the other methods as well

theertha120 commented on src/main/java/casio.java:
```diff
@@ -0,0 +1,127 @@
+        int task_number = 0;
+
+        //loop to get inputs
+        while (true) {
```
I like how you used different methods for the different user commands, makes the if-else if block neat

theertha120 commented on src/main/java/casio.java:
```diff
@@ -0,0 +1,127 @@
+                task_number++;
+            }
+        }
+        scanner.close();
```
Is this line required?

theertha120 commented on src/main/java/casio.java:
```diff
@@ -0,0 +1,127 @@
+    }
+
+    public static void add_task(Task[] task_array, String name, String type, int task_number, String details){
+        Task t = new Task(name, type);
```
You can also replace 'name' and 'type' with a more informative and longer parameter name

theertha120 reviewed (commented):
The code has a good structure, just a few naming conventions to fix to improve coding standard, and the code quality looks good overall!

louistaii committed:
Improve OOP and variable names

Classes Todo, Deadline and Event now inherit from class Tasks.
Changed variable names to suit naming convention

JudHoka commented on src/main/java/Casio.java:
```diff
@@ -0,0 +1,140 @@
+
+public class Casio{
+
+    private static final int MAX_SIZE = 100;
```
Awesome! as how the convention refers, constant must be all upper case

JudHoka commented on src/main/java/Casio.java:
```diff
@@ -0,0 +1,140 @@
+    }
+
+
+    public static void addTask(Task[] taskArray, String name){
```
Great job with the camelCase for the methods!

JudHoka commented on src/main/java/Task.java:
```diff
@@ -1,30 +1,22 @@
-{
+public class Task {
     protected String description;
     protected boolean isDone;
```
Yes, booleans have to sound like an action verb, well done!

JudHoka commented on src/main/java/Casio.java:
```diff
@@ -0,0 +1,140 @@
+                    System.out.println("Please insert a valid end name.");
+                }
+            } else if (input.startsWith("deadline ")){
+                String[] parts = input.substring(9).split("/by");
```
Arrays do have to sound like a collection of items, great job for the initiative for using plural!

JudHoka commented on src/main/java/Casio.java:
```diff
@@ -0,0 +1,140 @@
+    }
+
+    public static void printTasks(Task[] taskArray){
+        for (int i = 0; i < taskNumber; i++) {
```
Good implemention of for loop variables using i!

louistaii committed:
Add Parser and UI class

To better divide work between classes

louistaii committed:
Add error handling for exceptions

Handled errors:
invalidIndex
missingDetails
missingTaskName
Invalid commands
Repeated marking/unmarking tasks

louistaii committed:
Merge branch 'master' into branch-Level-5

louistaii committed:
Merge pull request #1 from louistaii/branch-Level-5

Add error handling for exceptions

louistaii committed:
Sort classes into packages

Refactored code for packaging

louistaii committed:
Add delete task functionality and convert array to ArrayList

Added delete function to delete tasks

Changed taskArray from an array to an ArrayList for better functionalities

louistaii committed:
Add support for saving tasks to hard disk

Tasks are now saved in ./data/casio.txt whenever added/marked/unmarked

louistaii committed:
Merge branch 'branch-Level-6'

louistaii committed:
Dummy commit to allow merge

To fix error when merging previously

louistaii committed:
Add save tasks function

louistaii committed:
Merge branch 'branch-Level-7' of https://github.com/louistaii/ip into branch-Level-7

louistaii committed:
Fix merging

louistaii committed:
Merge branch 'branch-Level-7'

xenthm commented on src/main/java/Casio/Casio.java:
```diff
@@ -0,0 +1,86 @@
+    public static void addTodo(String name){
+        Todo t = new Todo(name);
+        taskArray.add(t);
+        System.out.println("Added todo: "+ name);
```
Do try to use the formatting for white space outlined in [the Java coding standards for this course](https://se-education.org/guides/conventions/java/index.html#layout) in `"Added todo: "+ name`. I point out this line to avoid repeating myself -- it happens in other parts of your codebase too.

xenthm commented on src/main/java/Casio/tasks/TaskStorage.java:
```diff
@@ -0,0 +1,135 @@
+        }
+    }
+
+    public static void appendFile(Task task) {
```
I understand where you are coming from with the name `appendFile`, but consider making it more specific (i.e. `appendTaskToFile`) in the spirit of [using the method name to explain](https://nus-cs2113-ay2425s2.github.io/website/se-book-adapted/chapters/codeQuality.html#use-name-to-explain) what it does.

xenthm commented on src/main/java/Casio/exceptions/CasioException.java:
```diff
@@ -0,0 +1,20 @@
+        super(message);
+    }
+
+    public static void invalidIndex(int index, int taskNumber) throws CasioException {
+        index++;
+        throw new CasioException("OOPS! There are no tasks with index: " + index + " (valid index: 1 to " +taskNumber+")");
+    }
+
+    public static void missingDetails(String detail, String name) throws CasioException {
+        throw new CasioException("OOPS! Missing details ("+detail+") for: " + name + " (eg. '"+ name+ " /"+detail+" InsertDetailHere')");
+    }
+
+    public static void missingTaskName(String type) throws CasioException {
+        throw new CasioException("OOPS! You must specify a name for " + type + " tasks! (eg. '"+type+" InsertNameHere')");
+    }
```
Great job differentiating your exceptions. However, you can consider making even more custom exceptions that extend `CasioException` to better differentiate them for debugging. Your method, while correct, does not allow you to check which of the 3 custom exceptions you defined here occurred (other than using the printed exception message).

xenthm commented on src/main/java/Casio/Casio.java:
```diff
@@ -0,0 +1,86 @@
+        Scanner scanner = new Scanner(System.in);
+
+        //loop to get inputs
+        while (true) {
+            String input = scanner.nextLine();
+            taskNumber = Parser.parseInput(input, taskNumber, taskArray);
+        }
```
Correct me if I'm wrong, but it seems like there is no way to exit the program (even by giving the `bye` command) since the parser just calls `Ui.exit()`. This `while` loop continues executing. Is this the intended behaviour?

louistaii committed:
Refactor code style, fix exit bug, improve exceptions and naming

Refactored code to follow spacing convention
Exit function now exits loop
Naming of functions made clearer
Added more exceptions for easier debugging

louistaii committed:
Improve UI's appearance and interaction

louistaii committed:
Add class TaskList

Made the code more OOP

louistaii committed:
Add dates and times support

List can now be sorted by deadline / event start time (earliest to latest)

louistaii committed:
Add find function

louistaii committed:
Merge branch 'master' of https://github.com/louistaii/ip into branch-Level-8

louistaii committed:
Merge pull request #2 from louistaii/branch-Level-8

Add dates and times support

louistaii committed:
Merge branch 'master' of https://github.com/louistaii/ip into branch-Level-9

louistaii committed:
Merge pull request #3 from louistaii/branch-Level-9

Add find function

louistaii committed:
Rename Parser class for more clarity

louistaii committed:
Add JavaDoc for methods and classes

louistaii committed:
Merge branch 'master' of https://github.com/louistaii/ip into A-JavaDoc

louistaii committed:
Merge pull request #4 from louistaii/A-JavaDoc

Add JavaDoc for methods and classes

louistaii committed:
Add User Guide

louistaii committed:
Add sort by name, filter by task type methods. Update user guide

louistaii committed:
Add tests, update README

louistaii committed:
Fix UI bug with symbols, change tests to reflect change in UI

louistaii committed:
Update README.md
